### DISCUSSION TITLE
Split panes incorrectly restored when using neovide

### DISCUSSION THREAD

**Discussion by humblehacker** [rmagatti/auto-session#270] [Q&A] [answered]
Auto-session works great most of the time, but there's one area I have a persistent problem. When I open neovim using neovide with auto_restore_enabled = true, all my split panes are squashed into the top left of the window. This problem doesn't occur if I turn off auto-restore and manually restore with :SessionRestore, so I suspect it has something to do with the initial neovide window being somehow very small at the time that auto-restore is executed. I've tried messing around with delaying auto-restore, but I'm not well-versed in the intricacies of neovim/neovide startup (using lazy.nvim btw) so I've so far been unsuccessful.

To be clear, I don't think there's anything wrong with auto-session. I'm only hoping someone here might suggest some workarounds I could try to improve this situation. Thanks!

**Comment by rmagatti** [comment]
Yeah this sounds like a session is being restored before some kind of window events. I suppose what you could try would be to turn off auto restore enabled and set up a delayed auto command to restore the session? Not sure how that would look like to be honest but may be worth a try. I use neovide too but I only ever open it from the terminal already on the right project

**Reply by humblehacker** [reply]
> I use neovide too but I only ever open it from the terminal already on the right project

That's interesting. I do the same, but you're not experiencing the same issue? That would seem to suggest that it's something in my configuration that's causing the issue to show up. I'll try disabling all my plugins except auto-session and see what happens.

**Reply by humblehacker** [reply]
No luck. I've disabled everything except lazy.nvim and auto-session. The problem persists. So there's some other difference in our configurations that's accounting for the different behavior.

Partial success! I found that you've been publishing your dotfiles, so I grabbed your init.lua, stuffed it into my config, commented out a few things, and now when I restart my panes are in the right places! Interestingly, I can see neovide flash for a moment with the panes in their squashed positions, but then they reset to the correct positions.

So now I'm paring it down to find out why your config works and mine does not.

**Reply by humblehacker** [reply, marked as answer]
Found it!

```lua
vim.cmd [[
  augroup AutoAdjustResize
    autocmd!
    autocmd VimResized * execute "normal! \<C-w>="
  augroup end
]]
```

Looks like it's doing automatically what I've been doing manually for a while now - typing `⌃w=` when neovide starts.  Found an explanation here: https://www.reddit.com/r/neovim/comments/mv7cnz/comment/gvai9fp/
